test/builder/pipelineRun_test.go

```diff
@@ -77,3 +79,151 @@ func Test_PipelineRunAbort(t *testing.T) {
+				}},
+		},
+	}, pipelineRun.Spec.Logging)
+}
+
+func Test_PipelineRunLoggingIndexURL(t *testing.T) {
+	pipelineRun := PipelineRun("prefix1", "namespace1",
+		PipelineRunSpec(
+			LoggingWithIndexURL("testURL"),
+		),
+	)
+	assert.DeepEqual(t, &api.Logging{
+		Elasticsearch: &api.Elasticsearch{
+			IndexURL: "testURL",
+		},
+	}, pipelineRun.Spec.Logging)
+}
+
+func Test_PipelineRunLoggingWithRunIDAndWithIndexURL(t *testing.T) {
+	buildID := uuid.New().String()
+	realmID := uuid.New().String()
+	pipelineRun := PipelineRun("prefix1", "namespace1",
+		PipelineRunSpec(
+			LoggingWithIndexURL("testURL"),
+			LoggingWithRunID(&api.CustomJSON{
+				map[string]string{
+					"jobId":   "1",
+					"buildId": buildID,
+					"realmId": realmID,
+				}}),
+		),
+	)
+	assert.DeepEqual(t, &api.Logging{
+		Elasticsearch: &api.Elasticsearch{
+			IndexURL: "testURL",
+			RunID: &api.CustomJSON{
+				map[string]string{
+					"jobId":   "1",
+					"buildId": buildID,
+					"realmId": realmID,
+				}},
+		},
+	}, pipelineRun.Spec.Logging)
+}
+
+func Test_PipelineRunLoggingWithAllParams(t *testing.T) {
+	buildID := uuid.New().String()
+	realmID := uuid.New().String()
+	pipelineRun := PipelineRun("prefix1", "namespace1",
+		PipelineRunSpec(
+			LoggingWithIndexURL("testURL"),
+			LoggingWithRunID(&api.CustomJSON{
+				map[string]string{
+					"jobId":   "1",
+					"buildId": buildID,
+					"realmId": realmID,
+				}}),
+		),
+	)
+	assert.DeepEqual(t, &api.Logging{
+		Elasticsearch: &api.Elasticsearch{
+			IndexURL: "testURL",
+			RunID: &api.CustomJSON{
+				map[string]string{
+					"jobId":   "1",
+					"buildId": buildID,
+					"realmId": realmID,
+				}},
+		},
+	}, pipelineRun.Spec.Logging)
+}
+
+func Test_CheckConflictsBetweenIndexURLsOfTests(t *testing.T) {
+	buildID := uuid.New().String()
+	realmID := uuid.New().String()
+	testCases := []struct {
+		name           string
+		pipelineRun    *api.PipelineRun
+		expectedResult *api.Logging
+	}{
+		{
+			name: "pipelineRun with runID, indexURL and credential",
+			pipelineRun: PipelineRun("prefix1", "namespace1",
+				PipelineRunSpec(
+					LoggingWithIndexURL("foo"),
+					LoggingWithRunID(&api.CustomJSON{
+						map[string]string{
+							"jobId":   "1",
+							"buildId": buildID,
+							"realmId": realmID,
+						}}),
+				),
+			),
+			expectedResult: &api.Logging{
+				Elasticsearch: &api.Elasticsearch{
+					IndexURL: "foo",
+					RunID: &api.CustomJSON{
+						map[string]string{
+							"jobId":   "1",
+							"buildId": buildID,
+							"realmId": realmID,
+						}},
+				},
+			},
+		},
+		{
+			name: "pipelineRun with different indexURL than previous test",
+			pipelineRun: PipelineRun("prefix1", "namespace1",
+				PipelineRunSpec(
+					LoggingWithIndexURL("bar"),
+				),
+			),
+			expectedResult: &api.Logging{
+				Elasticsearch: &api.Elasticsearch{
+					IndexURL: "bar",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
```suggestion
                       tc := tc
			t.Parallel()
```
Avoid mismatch in test

Reply:
done